src/catalog/catalog_entry/scalar_function_catalog_entry.cpp

```diff
@@ -5,6 +5,10 @@
 namespace kuzu {
 namespace catalog {
 
+ScalarFunctionCatalogEntry::ScalarFunctionCatalogEntry(
```

Comment:
I don't think you need an additional constructor for const char* name. implicit cast will happen when the user provides name as a const char*.